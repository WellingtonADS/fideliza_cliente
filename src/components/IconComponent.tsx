import React from 'react';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import { View, Text, StyleSheet, ViewStyle, TextStyle } from 'react-native';
import { AppIcons, AppIconKey } from './iconNames';

// Tipagem dos props seguindo o padrão usado em fideliza_gestao
export type IconComponentProps = {
  name?: string;         // Nome direto FontAwesome (prioridade sobre 'icon')
  icon?: AppIconKey;     // Chave semântica do mapeamento AppIcons
  size?: number;         // Tamanho (default 30)
  color?: string;        // Cor (default #000)
  label?: string;        // Rótulo opcional abaixo do ícone
  containerStyle?: ViewStyle; // Estilo extra para o container
  labelStyle?: TextStyle;     // Estilo extra para o texto
};

const IconComponent: React.FC<IconComponentProps> = ({
  name,
  icon,
  size = 30,
  color = '#000',
  label,
  containerStyle,
  labelStyle,
}) => {
  const resolvedName = name || (icon ? AppIcons[icon] : 'question');
  return (
    <View style={[styles.container, containerStyle]}>
      <FontAwesome name={resolvedName} size={size} color={color} />
      {label && <Text style={[styles.label, labelStyle]}>{label}</Text>}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  label: {
    marginTop: 5,
    fontSize: 14,
    color: '#333',
  },
});

export default IconComponent;
