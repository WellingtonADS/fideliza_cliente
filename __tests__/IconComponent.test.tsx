import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import IconComponent from '../src/components/IconComponent';

// Mock da lib de ícones: componente funcional simples que não acessa variáveis externas
jest.mock('react-native-vector-icons/FontAwesome', () => {
  return function MockFontAwesome() {
    return null;
  };
});

describe('IconComponent (semantic mapping)', () => {
  it('renderiza com chave semântica', () => {
    let tree: any;
    ReactTestRenderer.act(() => {
      tree = ReactTestRenderer.create(
        <IconComponent icon="home" size={20} color="#123" label="Início" />,
      );
    });
    expect(tree.toJSON()).toMatchInlineSnapshot(`
      <View
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#333",
                "fontSize": 14,
                "marginTop": 5,
              },
              undefined,
            ]
          }
        >
          Início
        </Text>
      </View>
    `);
  });

  it('prioriza prop name sobre icon', () => {
    let tree: any;
    ReactTestRenderer.act(() => {
      tree = ReactTestRenderer.create(
        <IconComponent name="user" icon="home" />,
      );
    });
    expect(tree.toJSON()).toMatchInlineSnapshot(`
      <View
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      />
    `);
  });

  it('usa ícone fallback quando nenhum é informado', () => {
    let tree: any;
    ReactTestRenderer.act(() => {
      tree = ReactTestRenderer.create(<IconComponent />);
    });
    expect(tree.toJSON()).toMatchInlineSnapshot(`
      <View
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      />
    `);
  });
});
