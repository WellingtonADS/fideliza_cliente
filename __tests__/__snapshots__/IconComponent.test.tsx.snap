// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconComponent (semantic mapping) prioriza prop name sobre icon 1`] = `null`;

exports[`IconComponent (semantic mapping) renderiza com chave semântica 1`] = `null`;

exports[`IconComponent (semantic mapping) usa ícone fallback quando nenhum é informado 1`] = `null`;
